pyeudiw/satosa/backend.py

```diff
@@ -215,14 +232,43 @@ def redirect_endpoint(self, context, *args):
             raise NoBoundEndpointError("request_uri not valid")
 
         # take the encrypted jwt, decrypt with my public key (one of the metadata) -> if not -> exception
-
+        
+        jwt = context.request["response"]
+        jwk = JWK(self.config["federation"]["federation_jwks"][0], "RSA")
+        
+        jweHelper = JWEHelper(jwk)
+        
+        decrypted_data = jweHelper.decrypt(jwt)
+        
         # get state and nonce, do lookup on the db -> if not -> exception
-
+        
+        nonce = decrypted_data.get("nonce", None) #TODO Fix this field handling
+        state = decrypted_data.get("state", None) #TODO Fix this field handling
+        
         # check with pydantic on the JWT schema
-
+                
+        ResponseValidator(**decrypted_data)
+        
         # check if vp_token is string or array, if array iter all the elements
-
+        
         # take the single vp token, take the credential within it, use cnf.jwk to validate the vp token signature -> if not exception
+        
+        vp_token = decrypted_data["vp_token"]
+        
+        claims = []
+        if type(vp_token) is str:
+            valid, value = self._check_vp_token(vp_token)
```

Comment:
check_vp_token should be in `pyeudiw.openid4vp` module, and used in the satosa backend